x/gov/client/cli/query_test.go

```diff
@@ -0,0 +1,427 @@
+			}
+		})
+	}
+}
+
+func (s *CLITestSuite) TestCmdParam() {
+	testCases := []struct {
+		name         string
+		args         []string
+		expCmdOutput string
+	}{
+		{
+			"voting params",
+			[]string{
+				"voting",
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+			},
+			`voting --output=json`,
+		},
+		{
+			"tally params",
+			[]string{
+				"tallying",
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+			},
+			`tallying --output=json`,
+		},
+		{
+			"deposit params",
+			[]string{
+				"deposit",
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+			},
+			`deposit --output=json`,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			cmd := cli.GetCmdQueryParam()
+			cmd.SetArgs(tc.args)
+
+			if len(tc.args) != 0 {
+				s.Require().Contains(fmt.Sprint(cmd), strings.TrimSpace(tc.expCmdOutput))
+			}
+		})
+	}
+}
+
+func (s *CLITestSuite) TestCmdProposer() {
+	testCases := []struct {
+		name         string
+		args         []string
+		expCmdOutput string
+	}{
+		{
+			"without proposal id",
+			[]string{
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+			},
+			"--output=json",
+		},
+		{
+			"json output",
+			[]string{
+				"1",
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+			},
+			"1 --output=json",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			cmd := cli.GetCmdQueryProposer()
+			cmd.SetArgs(tc.args)
+
+			if len(tc.args) != 0 {
+				s.Require().Contains(fmt.Sprint(cmd), strings.TrimSpace(tc.expCmdOutput))
+			}
+		})
+	}
+}
+
+func (s *CLITestSuite) TestCmdTally() {
+	testCases := []struct {
+		name         string
+		args         []string
+		expCmdOutput string
+	}{
+		{
+			"without proposal id",
+			[]string{
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+			},
+			"--output=json",
+		},
+		{
+			"json output",
+			[]string{
+				"2",
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+			},
+			"2 --output=json",
+		},
+		{
+			"json output",
+			[]string{
+				"1",
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+			},
+			"1 --output=json",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			cmd := cli.GetCmdQueryTally()
+			cmd.SetArgs(tc.args)
+
+			if len(tc.args) != 0 {
+				s.Require().Contains(fmt.Sprint(cmd), strings.TrimSpace(tc.expCmdOutput))
+			}
+		})
+	}
+}
+
+func (s *CLITestSuite) TestCmdGetProposal() {
+	testCases := []struct {
+		name         string
+		args         []string
+		expCmdOutput string
+	}{
+		{
+			"get non existing proposal",
+			[]string{
+				"10",
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+			},
+			"10 --output=json",
+		},
+		{
+			"get proposal with json response",
+			[]string{
+				"1",
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+			},
+			"1 --output=json",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		s.Run(tc.name, func() {
+			cmd := cli.GetCmdQueryProposal()
+			cmd.SetArgs(tc.args)
+
+			if len(tc.args) != 0 {
```

Comment:
I don't really understand why you need this for tests where no test case have no argument.

Reply:
that's true, but when I talked to marko in the initial phase of these tests, he said when there aren't any args to the cmds, we should atleast check if the cmd with some args or flags are passed correctly to it.

Reply:
Right, but this isn't reflected here. Now, if there is no argument, it tests nothing.

Reply:
but we are still passing some flag args [here](e.g., https://github.com/cosmos/cosmos-sdk/pull/13717/files/c36b001952e255742770a02b3470bfdcd96ec2a6#diff-c112c9abababa7f43d1588715ebcde623a3e03e3c1dce8881ed9c702e4755442R21)
shouldn't we consider this case ?

Reply:
Yes we should, what I mean is in some tests there is always an argument. So in some tests, condition isn't needed.
f.e. https://github.com/cosmos/cosmos-sdk/pull/13717/files/c36b001952e255742770a02b3470bfdcd96ec2a6#diff-c112c9abababa7f43d1588715ebcde623a3e03e3c1dce8881ed9c702e4755442R84-R86

Reply:
my bad for not understanding your comment correctly, will make required changes.
thanks @julienrbrt